src/utils/request-utils.ts

```diff
@@ -42,8 +42,31 @@ export const formDataToQuery = function (formdata: Record<string, any> | FormDat
 }
 
 export const getQueryParam = function (url: string, param: string): string {
+    // Check if URL and URLSearchParams are available in the global scope
+    // Fall back to manual parsing if URL/URLSearchParams aren't supported
+    if (typeof URL === 'undefined' || typeof URLSearchParams === 'undefined') {
+        return getQueryParamFallback(url, param)
+    }
+
+    // Disabling compat/compat here because we've checked its existence above
+    const urlObj = new URL(url) // eslint-disable-line compat/compat
+
+    // Remove leading ? from search string if present
+    // This is not really necessary if there's a single ? in the search string
+    // but it's helpful if someone has some malformed URL with multiple ?s
+    const search = urlObj.search.replace(/^\?*/g, '')
+    const searchParams = new URLSearchParams(search) // eslint-disable-line compat/compat
+
+    return searchParams.get(param) ?? ''
+}
```

Comment:
Open to feedback on whether this is needed. I think using the builtins are better than rolling our own code like we have on the fallback, but happy to be told otherwise and just stick to the previous thing we had going

Also, I'm not decoding the params anymore, so I might need to add that back 🤔

Reply:
I'd rather only have 1 version. Bundle size is important. Sorry man =/ I'd love to kill off the old version and only have one that uses built-in methods, but if we need to support older browsers then we can't do that.

Reply:
Not a problem, will revert